pkg/bmc/access_test.go

```diff
@@ -11,162 +11,192 @@ func init() {
-	T, H, P, A, err := getTypeHostPort("my.favoritebmc.com")
+	url, err := getParsedURL("my.favoritebmc.com")
 	if err != nil {
 		t.Fatalf("unexpected parse error: %v", err)
 	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
+	if url.Scheme != "ipmi" {
+		t.Fatalf("unexpected type: %q", url.Scheme)
+	}
+	if url.Port() != "" { // default is set in DriverInfo() method
+		t.Fatalf("unexpected port: %q", url.Port())
 	}
-	if P != "" { // default is set in DriverInfo() method
-		t.Fatalf("unexpected port: %q", P)
+	if url.Hostname() != "my.favoritebmc.com" {
+		t.Fatalf("unexpected hostname: %q", url.Hostname())
 	}
-	if H != "my.favoritebmc.com" {
-		t.Fatalf("unexpected hostname: %q", H)
+	if url.Path != "" {
+		t.Fatalf("unexpected path: %q", url.Path)
 	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
+	if len(url.Query()) != 0 {
+		t.Fatalf("unexpected query length: %q", len(url.Query()))
 	}
 }
 
 func TestParseHostPort(t *testing.T) {
-	T, H, P, A, err := getTypeHostPort("192.168.122.1:6233")
+	url, err := getParsedURL("192.168.122.1:6233")
 	if err != nil {
 		t.Fatalf("unexpected parse error: %v", err)
 	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
+	if url.Scheme != "ipmi" {
+		t.Fatalf("unexpected type: %q", url.Scheme)
 	}
-	if H != "192.168.122.1" {
-		t.Fatalf("unexpected hostname: %q", H)
+	if url.Hostname() != "192.168.122.1" {
+		t.Fatalf("unexpected hostname: %q", url.Hostname())
 	}
-	if P != "6233" {
-		t.Fatalf("unexpected port: %q", P)
+	if url.Port() != "6233" {
+		t.Fatalf("unexpected port: %q", url.Port())
 	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
+	if url.Path != "" {
+		t.Fatalf("unexpected path: %q", url.Path)
+	}
+	if len(url.Query()) != 0 {
+		t.Fatalf("unexpected query length: %q", len(url.Query()))
 	}
 }
 
 func TestParseHostPortIPv6(t *testing.T) {
-	T, H, P, A, err := getTypeHostPort("[fe80::fc33:62ff:fe83:8a76]:6233")
+	url, err := getParsedURL("[fe80::fc33:62ff:fe83:8a76]:6233")
 	if err != nil {
 		t.Fatalf("unexpected parse error: %v", err)
 	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
+	if url.Scheme != "ipmi" {
+		t.Fatalf("unexpected type: %q", url.Scheme)
+	}
+	if url.Hostname() != "fe80::fc33:62ff:fe83:8a76" {
+		t.Fatalf("unexpected hostname: %q", url.Hostname())
 	}
-	if H != "fe80::fc33:62ff:fe83:8a76" {
-		t.Fatalf("unexpected hostname: %q", H)
+	if url.Port() != "6233" {
+		t.Fatalf("unexpected port: %q", url.Port())
 	}
-	if P != "6233" {
-		t.Fatalf("unexpected port: %q", P)
+	if url.Path != "" {
+		t.Fatalf("unexpected path: %q", url.Path)
 	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
+	if len(url.Query()) != 0 {
+		t.Fatalf("unexpected query length: %q", len(url.Query()))
 	}
 }
 
 func TestParseHostNoPortIPv6(t *testing.T) {
-	// They either have to give us a port or a URL scheme with IPv6.
-	_, _, _, _, err := getTypeHostPort("[fe80::fc33:62ff:fe83:8a76]")
+	// TBy default we fallback to IPMI, also with IPv6.
```

Comment:
typo, but also this new comment doesn't seem to match the code.

Reply:
thanks